spec/factories/subjects/secondary_subjects.rb

```diff
@@ -0,0 +1,197 @@
+FactoryBot.define do
+  subjects = [
+      # Modern Lanuages has been removed as it would cause validation errors if it
+      # isn't paired with a valid modern language subject use a trait if you want to create it
+      { subject_name: "Art and design", subject_code: "W1" },
+      { subject_name: "Science", subject_code:  "F0" },
+      { subject_name: "Biology", subject_code:  "C1" },
+      { subject_name: "Business studies", subject_code: "08" },
+      { subject_name: "Chemistry", subject_code: "F1" },
+      { subject_name: "Citizenship", subject_code:  "09" },
+      { subject_name: "Classics", subject_code: "Q8" },
+      { subject_name: "Communication and media studies", subject_code: "P3" },
+      { subject_name: "Computing", subject_code: "11" },
+      { subject_name: "Dance", subject_code: "12" },
+      { subject_name: "Design and technology", subject_code: "DT" },
+      { subject_name: "Drama", subject_code: "13" },
+      { subject_name: "Economics", subject_code: "L1" },
+      { subject_name: "English", subject_code: "Q3" },
+      { subject_name: "Geography", subject_code: "F8" },
+      { subject_name: "Health and social care", subject_code: "L5" },
+      { subject_name: "History", subject_code:  "V1" },
+      { subject_name: "Mathematics", subject_code: "G1" },
+      { subject_name: "Music", subject_code: "W3" },
+      { subject_name: "Philosophy", subject_code: "P1" },
+      { subject_name: "Physical education", subject_code: "C6" },
+      { subject_name: "Physics", subject_code: "F3" },
+      { subject_name: "Psychology", subject_code: "C8" },
+      { subject_name: "Religious education", subject_code: "V6" },
+      { subject_name: "Social sciences", subject_code: "14" },
+    ]
+
+  factory :secondary_subject do
+    transient do
+      sample_subject { subjects.sample }
+    end
+
+    subject_name { sample_subject[:subject_name] }
+    subject_code { sample_subject[:subject_code] }
+    type { :SecondarySubject }
+
+    trait :art_and_design do
+      subject_name { "Art and design" }
+      subject_code { "W1" }
```

Comment:
Also worth using the association already defined here to pull out the subject code.